### DISCUSSION TITLE
store api not working with hooks api

### DISCUSSION THREAD

**Discussion by prbxr** [pmndrs/jotai#1829] [General] [answered]
Hi,

Seems that if setting store atom, and using atom from component, the component will not react, is  it planned ?

const store = createStore()
store.set(fooAtom, 'foo')

and then

useAtom(fooAtom)

**Comment by wootra** [comment]
I don't know full code, but useAtom has array form of return value, if you miss it. Also, you missed the {store: store} option. 
```
const [retVal, fetch] = useAtom(fooAtom, {store} ); // <-- you are connecting your atom with store.
```
and it should work

**Comment by prbxr** [comment]
I don't find that syntax - useAtom(fooAtom, {store} )

i have to import like this:

```
import { createStore, atom, getDefaultStore } from 'jotai/vanilla'
import { useAtom } from 'jotai'
```

so they are coming from different places 
and they have different typing

when I do this, it does not work because of typings
it complaints on the atom and on the store

```
import { createStore, atom, getDefaultStore } from 'jotai/vanilla'
import { useAtom } from 'jotai'

let att = atom(0)

getDefaultStore().set(att, (c) => c + 1)

let [vvv] = useAtom(att, { store: getDefaultStore() })
```

useAtom does not accept atom from vanilla import...

**Reply by dai-shi** [reply]
It works on TypeScript Playground: https://tsplay.dev/mAJA8W

**Comment by dai-shi** [comment, marked as answer]
> Hi,
> 
> Seems that if setting store atom, and using atom from component, the component will not react, is it planned ?
> 
> const store = createStore() store.set(fooAtom, 'foo')
> 
> and then
> 
> useAtom(fooAtom)

You should provide a store with Provider.

```tsx
<Provider store={store}>
```
then `useAtom(fooAtom)` works under the provider.

**Reply by wootra** [reply]
Not sure why type error is showing.. as you see this code, the atom type comes from jotai/vanilla. should not be the type of atom itself. maybe you're setting 'function' while your atom is typed as number.
```
let att = atom(0); // <==number type primitive atom
getDefaultStore().set(att, (c) => c + 1); <== you're setting a function in the number type atom.
```
you can do like this instead:
```
const store = getDefaultStore();
store.set(att, store.get(att) + 1);
```

**Comment by prbxr** [comment]
Yes, I tried it in codesandbox and it does not give type errors.
I upgraded locally to 2.0.3 and now it still gives me error for  this syntax { store: ... }
But ok, at least it works in codesandbox.
Actually here  in this codesandbox there is no need to specify a { store: ... } in useAtom, it just works with the default store.
When using a Provider no need to speficy a store in useAtom as well.
https://codesandbox.io/p/sandbox/nifty-curie-z8ixvg?file=%2FREADME.md
